options/flags.go

```diff
@@ -84,11 +84,14 @@ func (o *Options) AddFlags(cmd *cobra.Command) {
 	)
 
 	// TODO(v5): Should this be behind a feature flag?
-	cmd.Flags().StringVar(
+	cmd.PersistentFlags().StringVar(
 		&o.Commit,
 		FlagCommit,
 		o.Commit,
-		"commit to analyze",
+		"For the default scorecard run, this specifies a code commit to analyze. "+
+			`For dependencydiff, this includes the two commits BASE and HEAD, use "..." to separate them. `+
+			`Both commitSHAs (commit_A_SHA...commit_B_SHA) or branch names ("main...dev") or a mix of them `+
+			`(main...commit_A_SHA) are supported. Please don't use the default value for dependencydiff usage.`,
```

Comment:
I was considering not using the `--commit` flag to take the `base...head` dependnecy-diff input, since it is used for taking exactly one commitSHA used for the original Scorecard check; otherwise, it could be ambiguous if we use it differently in the root command and the `dependency-diff` sub-command.

Instead, maybe we could create two sub-command flags `--base` and `--head` for `dependency-diff` to take the base and head inputs. wdut? @laurentsimon @azeemshaikh38

Reply:
changed the flag usage: using the global flag `--commit` to receive `base...head` --> using two sub-command flags `--base` for `base` and `--head` for `head`